SpatialGDK/Source/SpatialGDKFunctionalTests/Private/TestMapGeneration.cpp

```diff
@@ -52,22 +54,36 @@ bool GenerateTestMaps()
 	TArray<UClass*> TestMapClasses;
 	for (TObjectIterator<UClass> Iter; Iter; ++Iter)
 	{
-		if (Iter->IsChildOf(UGeneratedTestMap::StaticClass()) && *Iter != UGeneratedTestMap::StaticClass())
+		if (Iter->IsChildOf(UGeneratedTestMap::StaticClass()) || Iter->IsChildOf(ASpatialFunctionalTest::StaticClass()))
 		{
 			TestMapClasses.Add(*Iter);
 		}
 	}
 
 	for (UClass* TestMapClass : TestMapClasses)
 	{
-		UGeneratedTestMap* TestMap = NewObject<UGeneratedTestMap>(GetTransientPackage(), TestMapClass);
-		TestMap->AddToRoot(); // Okay, must admit, not completely sure what's going on here, seems like even though the commandlet is
-							  // the outer of the newly generated object, the object still gets GCed when creating a new map, so have to add
-							  // to root here to prevent GC
+		IGeneratableTestMap* TestMap;
```

Comment:
Maybe all the function calls here could be made static? If possible it could make this code a lot simpler.
These functions:
```
ShouldGenerateMap()
SaveMap()
GenerateMap()
```

Reply:
I think its probably out of scope for this PR. But basically if all this can be made static with maybe a constructor like function we don't have to go through all this effort of keeping a temp instance around.